jsonrpc/eth_endpoint.go

```diff
@@ -320,6 +320,23 @@ func (e *Eth) BlockNumber() (interface{}, error) {
 	return argUintPtr(h.Number), nil
 }
 
+// SignTransaction sign transaction with key of account, key need to be unlocked
+func (e *Eth) SignTransaction(txn *txnArgs) (interface{}, error) {
+	account := accounts.Account{Address: *txn.From}
+
+	keyStore, err := getKeystore(e.accManager)
+	if err != nil {
+		return nil, err
+	}
+
+	tx, err := DecodeTxn(txn, e.store, true)
+	if err != nil {
+		return nil, err
+	}
+
+	return keyStore.SignTx(account, tx)
```

Comment:
If I'm not mistaken, the SignTransaction should return the RLP-encoded byte array of signed transaction (https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_signtransaction).

In fact, the Geth returns both the transaction instance and RLP-encoded byte array (in form of the `SignTransactionResult`): https://github.com/ethereum/go-ethereum/blob/a71f6f91fdb2fcccdd0c1e336267951d45932f26/internal/ethapi/api.go#L1892-L1931

Reply:
Fixed: https://github.com/Ethernal-Tech/blade/pull/290/commits/fb42eddc7de904e0ff2ffa517472f6aa16531ddd